src/components/partnership/partnership.service.ts

```diff
@@ -493,15 +511,45 @@ export class PartnershipService {
   async update(input: UpdatePartnership, session: ISession) {
     // mou start and end are now computed fields and do not get updated directly
     const object = await this.readOne(input.id, session);
+    let changes = input;
+    if (
+      !this.validateFundingType(
+        input.fundingType ?? object.fundingType.value,
+        input.types ?? object.types.value
+      )
+    ) {
+      changes = {
+        ...input,
+        fundingType: (null as unknown) as PartnershipFundingType,
+      };
+    }
 
-    const { mou, agreement, ...rest } = input;
+    if (input.fundingType && input.types) {
+      if (!this.validateFundingType(input.fundingType, input.types)) {
+        await this.db.sgUpdateProperties({
+          session,
+          object,
+          props: ['types', 'fundingType'],
+          changes: {
+            types: [],
+            fundingType: (null as unknown) as PartnershipFundingType,
+          },
+          nodevar: 'partnership',
+        });
+        // Throw User Input Error
+        this.verifyFundingType(input.fundingType, input.types);
+      }
+    }
```

Comment:
I don't understand all these changes. Why call validateFundingType twice? Why update types & fundingType separately from the rest of the properties below?

Reply:
My steps:
- If the `fundingType` and `types` are passed, and they are invalid, then set those fields to [], `null` and throw `InputException`
- In other cases, it only set the `fundingType` to `null` following other fields updated.

Reply:
`this.verifyFundingType(input.fundingType, input.types);` is just for throw error

Reply:
If the input is invalid we can just throw there. There's no reason to do a set before for that.